vms/platformvm/block/builder/builder.go

```diff
@@ -308,6 +292,85 @@ func buildBlock(
 	)
 }
 
+func packBlockTxs(mempool mempool.Mempool, backend *txexecutor.Backend, manager blockexecutor.Manager, timestamp time.Time) ([]*txs.Tx, error) {
+	preferredID := manager.Preferred()
+	stateDiff, err := state.NewDiff(preferredID, manager)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Shouldn't we be passing these in from `buildBlock`?

Reply:
Yes